File: app/src/main/java/br/com/patrocine/patrocine/ui/activities/ProductDetailActivity.java
package br.com.patrocine.patrocine.ui.activities;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import br.com.patrocine.patrocine.BuildConfig;
import br.com.patrocine.patrocine.R;
import br.com.patrocine.patrocine.model.Pizza;
import br.com.patrocine.patrocine.model.Snack;
import com.squareup.picasso.Picasso;

public class ProductDetailActivity extends AppCompatActivity {

    private String title;
    private String ingredientes;
    private String preco;

    ImageView productImage;
    TextView tvProductTitle;
    TextView tvProductIngredients;
    TextView tvProductValue;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_product_detail);

        productImage = findViewById(R.id.productImage);
        tvProductTitle = findViewById(R.id.tvProductTitle);
        tvProductIngredients = findViewById(R.id.tvProductIngredients);
        tvProductValue = findViewById(R.id.tvProductValue);

        Bundle b = getIntent().getExtras();

        String title = getResources().getString(R.string.app_name);
        setTitle(title);

        if(b.getString("type").equals("PIZZAS")){
            Pizza p = (Pizza) b.getSerializable("product");
            tvProductTitle.setText("PIZZA " + p.getTitle());
            tvProductIngredients.setText(p.getIngredients());
            tvProductValue.setText("R$ "+ p.getPrice());
            String imageUrl = BuildConfig.img_server + "pizzas/" + p.getPreview();
        }

        if(b.getString("type").equals("LANCHES")){
            Snack s = (Snack) b.getSerializable("product");
            tvProductTitle.setText(s.getTitle());
            tvProductIngredients.setText(s.getIngredients());
            tvProductValue.setText("R$ "+ s.getPrice());
            String imageUrl = s.getPreview();

            // Picasso.with(this).load(imageUrl).into(productImage);
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_product_detail, menu);
        return false;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onBackPressed() {
        // finaliza a activity quando clicar em voltar
        finish();
    }
}

File: app/src/main/java/br/com/patrocine/patrocine/ui/activities/ProductDetailActivity.kt
package br.com.patrocine.patrocine.ui.activities

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import br.com.patrocine.patrocine.BuildConfig
import br.com.patrocine.patrocine.R
import br.com.patrocine.patrocine.model.Pizza
import br.com.patrocine.patrocine.model.Snack
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.activity_product_detail.*

class ProductDetailActivity : AppCompatActivity() {

    private val title: String? = null
    private val ingredientes: String? = null
    private val preco: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_product_detail)

        val b = intent.extras

        val title = resources.getString(R.string.app_name)
        setTitle(title)

        if (b!!.getString("type") == "PIZZAS") {
            val p = b.getSerializable("product") as Pizza
            tvProductTitle.text = "PIZZA " + p.title!!
            tvProductIngredients.text = p.ingredients
            tvProductValue.text = "R$ " + p.price!!
            val imageUrl = p.preview
        }

        if (b.getString("type") == "LANCHES") {
            val s = b.getSerializable("product") as Snack
            tvProductTitle.text = s.title
            tvProductIngredients.text = s.ingredients
            tvProductValue.text = "R$ " + s.price!!
            val imageUrl = s.preview

            Picasso.get().load(imageUrl).into(productImage)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_product_detail, menu)
        return false
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }

    override fun onBackPressed() {
        // finaliza a activity quando clicar em voltar
        finish()
    }
}
